components/sup/src/manager/service/hooks.rs

```diff
@@ -188,13 +192,24 @@ impl HookTable {
         });
     }
 
-    fn load_hook(&self, hook_type: HookType, service: &Service) -> Option<Hook> {
-        let template = hook_template_path(service, &hook_type);
-        let concrete = hook_path(service, &hook_type);
-        let (user, group) = util::users::get_user_and_group(&service.package)
-            .expect("Can't determine user:group");
-        match fs::metadata(&template) {
-            Ok(_) => Some(Hook::new(hook_type, template, concrete, user, group)),
+    fn load_hook<T>(&self,
+                    hook_type: HookType,
+                    runtime_cfg: &RuntimeConfig,
+                    hooks: T,
+                    templates: T)
+                    -> Option<Hook>
+        where T: AsRef<Path>
```

Comment:
Just a note that with type `T`, both `hooks` and `templates` must be the same input type (i.e. both `&str`, `String`, etc.) otherwise you need a generic per-input ([like here](https://github.com/habitat-sh/habitat/blob/57a51506f486211c7db1668208ac2c5e7c5bd0fc/components/common/src/command/package/install.rs#L67-L68)).